### PR TITLE
Bump Gateway API to v0.7.1

### PR DESCRIPTION
**Pull request by sunjayBhatia**
To ensure compatibility with upcoming release

Fixes: #5336
Fixes: #5414

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
Will update dependency tag/PR title as new rc/GA versions become available

**Comment by sunjayBhatia** [comment]
Re-running to see if the TLSRoute test failure was a flake

Otherwise, we have the ObservedGeneration bump test failing and more redirect tests failing

**Comment by sunjayBhatia** [comment]
made some updates to the comment above, will have a PR to fix GW API tests coming shortly

File: .github/workflows/prbuild.yaml [hunk 1]
```diff
@@ -76,6 +76,8 @@ jobs:
           ./hack/actions/install-kubernetes-toolchain.sh $GITHUB_WORKSPACE/bin
           echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
       - name: generate
+        env:
```
Comment (sunjayBhatia) [inline .github/workflows/prbuild.yaml:79]: will undo changes here once we have another tagged patch release

**Comment by sunjayBhatia** [comment]
New failing test: `TestConformance/HTTPRouteHostnameIntersection/HTTPRoutes_that_do_intersect_with_listener_hostnames/1_request_to_'very.specific.com:1234/s1'_should_go_to_infra-backend-v1`

fails with error

`http.go:222: Response expectation failed for request: {URL: {Scheme:http Opaque: User: Host:172.18.255.205 Path:/s1 RawPath: OmitHost:false ForceQuery:false RawQuery: Fragment: RawFragment:}, Host: very.specific.com:1234, Protocol: HTTP, Method: GET, Headers: map[X-Echo-Set-Header:[]], UnfollowRedirect: false, Server: , CertPem: <truncated>, KeyPem: <truncated>}  not ready yet: expected host to be very.specific.com:1234, got very.specific.com (after 1µs)`

**Comment by sunjayBhatia** [comment]
Another failing test, looks like it was added after my review, see: https://github.com/kubernetes-sigs/gateway-api/pull/1984/files#r1200873543

**Comment by sunjayBhatia** [comment]
rebased on https://github.com/projectcontour/contour/pull/5390 to get the fixes from there

**Comment by sunjayBhatia** [comment]
looks like the webhook image isn't out yet

**Comment by sunjayBhatia** [comment]
Might have to undo https://github.com/projectcontour/contour/pull/3458 to get `TestGatewayConformance/HTTPRouteHostnameIntersection` passing again as the tests expect the port from the `Host` header is passed upstream and right now we're stripping it

**Comment by sunjayBhatia** [comment]
Filed https://github.com/kubernetes-sigs/gateway-api/issues/2091

**Comment by sunjayBhatia** [comment]
rebased on https://github.com/projectcontour/contour/pull/5437

**Comment by sunjayBhatia** [comment]
Conformance is all passing, just got to fix whatever is up with e2e tests from #5437

File: test/conformance/gatewayapi/gateway_conformance_test.go [hunk 2]
```diff
@@ -29,32 +34,45 @@ import (
...
 		// Keep the list of skipped features in sync with
 		// test/scripts/run-gateway-conformance.sh.
 		SkipTests: []string{
-			// These are skipped because the tests check for the
-			// original request port in the returned Location
+			// Test adds multiple HTTP Listeners to the Gateway with
+			// distinct ports. This is not supported in Contour until
+			// multi-Listener support is added.
+			// See: https://github.com/projectcontour/contour/issues/4960
+			tests.GatewayObservedGenerationBump.ShortName,
```
Comment (skriss) [inline test/conformance/gatewayapi/gateway_conformance_test.go:67]: I've confirmed that this test will pass with #5160, will update that PR to remove it from the skipped tests

**Review by skriss — APPROVED** [review]
🚀 thanks for all the work on this one @sunjayBhatia!

**Comment by skriss** [comment]
This PR still needs its own changelog file

**Pull request merged**